### ISSUE TITLE
Add support for Rancher in addition to DC/OS

### ISSUE THREAD

**Issue by akshaykarle** [JeroenSoeters/wheel#17]
When you run `wheel init`, it would be good if we can select DC/OS or rancher. Rancher is a similar orchestration platform to DC/OS. This will also mean introduce a yaml file that allows you to select the resources you'd like to manage with wheel.

I'd like to start by implementing the yaml file and the add rancher as a resource similar to DC/OS.

**Comment by JeroenSoeters** [comment]
Agreed, though I also think we should allow not installing a datacenter operating system at all, and running a container orchestration framework like k8 or Nomad directly on cloud providers like GPC or AWS. If Rancher abstracts away container orchestration, then we should maybe treat Rancher in Wheel as a container orchestration framework in itself. So Wheel would support deploying containers into Rancher, into Marathon, into k8 etc. Rancher might under the cover delegate to k8, Swarm or what not, but that is not a concern to Wheel anymore at that point. Would that work?

**Comment by akshaykarle** [comment]
Yes, rancher today supports swarm, k8s and cattle as the orchestration platforms.

Can we start fleshing out the implementation of the yaml file that allows specifing the resources to be managed by wheel or those that wheel should know about, like when rancher is already installed, we should be able to specify the rancher server url and the api keys to be used by wheel?

**Comment by JeroenSoeters** [comment]
That sounds good, wanna take a stab on it based on the one I have shared in Google drive? We can make sure that the format supports Rancher as well as k8 / AWS combination. I'm still looking into the Marathon/Mesos stuff. Can pair on it tomorrow.

**Comment by akshaykarle** [comment]
Sounds good. Will ping you to co-ordinate the time.
